src/mos.c

```diff
@@ -640,16 +640,60 @@ int mos_cmdSET(char * ptr) {
 // Returns:
 // - MOS error code
 //
-int	mos_cmdVDU(char *ptr) {
-	UINT24 	value;
-	
-	while(mos_parseNumber(NULL, &value)) {
-		if(value > 255) {
-			return 19;	// Bad Parameter
-		}
-		putch(value);
-	}
-	return 0;
+int mos_cmdVDU(char *ptr) {
+    char *value_str;
+    UINT24 value = 0;
+    
+    while (mos_parseString(NULL, &value_str)) {
+        UINT8 isLong = 0;
+        UINT8 base = 10;
+        char *endPtr;
+        size_t len = strlen(value_str);
+
+        //Strip semicolon notation and set as Long
+        if (len > 0 && value_str[len - 1] == ';') {
+            value_str[len - 1] = '\0';
+            len--;
+            isLong = 1;
+        }
+
+        // Check for '0x' or '0X' prefix
+        if (len > 2 && (value_str[0] == '0' && tolower(value_str[1] == 'x'))) {
+            base = 16;
+            value_str += 2;
+            len -= 2;

```

Comment:
I think these 2 lines should be removed and let strtol() call below to deal with this prefix. Otherwise, the parsing code will allow input like 0x0xFE to be parsed as a valid number.

Also, the independent checks for prefix/suffix should probably be converted to sequence of if / elseif. The idea is to accept as valid only numbers with at most one prefix or suffix, not any combination of all possible prefixes/suffixes.

Reply:
Agreed on those two lines, since strtol can handle 0x prefixes there's no need.

We discussed the second point and it costs us nothing to deal with situations where a user may double up on notation.